code/script.js

```diff
@@ -218,108 +223,190 @@ const generateBoard = () => {
           <button class="filled-button small" onclick="guess('${person.name}')">Guess</button>
         </div>
       </div>
-    `
-  })
-}
+    `;
+  });
+};
 
-// Randomly select a person from the characters array and set as the value of the variable called secret
+// Randomly select a person from the char-array and set as the value of the variable
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  secret =
+    charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+};
 
-// This function to start (and restart) the game
+// starts (and restarts) the game
+// setting charactersInPlay array to be all the characters to start with
+// resets the guess counter to 0 went reload/restart
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  charactersInPlay = CHARACTERS;
+  counterValue = 0;
+  updateDisplay(counterValue);
+  setSecret();
+  generateBoard();
+  winOrLose.style.display = "";
+};
+
+var counterVal = 0;
+
+function incrementClick() {
+  updateDisplay(++counterVal);
+}
+function updateDisplay(val) {
+  document.getElementById("counter-label").innerHTML = val;
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label;
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.value;
+  console.log("Questions value" + questions.value + category);
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
 
+  if (category === "hair" || category === "eyes") {
+    if (secret[category] === value) {
+      filterCharacters(true);
+      console.log("Yes, correct!");
```

Comment:
could remove the console logs if you wanted to shorten the code.

Reply:
oh yes, thanks, that was missed!